src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -110,58 +110,99 @@ public override int Execute()
+                {
+                    var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                    if (File.Exists(installStateFilePath))
                     {
-                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
-                        if (File.Exists(installStateFilePath))
+                        var installStateContents = InstallStateContents.FromPath(installStateFilePath);
+                        //  If install state has pinned workload set or manifest versions, then don't update workloads
+                        if (!string.IsNullOrEmpty(installStateContents.WorkloadVersion) || installStateContents.Manifests != null)
                         {
-                            var installStateContents = InstallStateContents.FromPath(installStateFilePath);
-                            _workloadSetVersion = installStateContents.WorkloadVersion;
+                            //  TODO: respect shouldUpdateWorkloads, or figure out update / install manifest difference in InstallingWorkloadCommand
+                            shouldUpdateWorkloads = false;
                         }
                     }
+                }
+
+                try
+                {
+                    Reporter.WriteLine();
 
                     DirectoryPath? offlineCache = string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption);
                     var workloadIds = _workloadIds.Select(id => new WorkloadId(id));
-                    if (string.IsNullOrWhiteSpace(_workloadSetVersion) && string.IsNullOrWhiteSpace(_workloadSetVersionFromGlobalJson))
+
+                    // Add workload Ids that already exist to our collection to later trigger an update in those installed workloads
+                    var installedWorkloads = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(_sdkFeatureBand);
+                    var previouslyInstalledWorkloads = installedWorkloads.Intersect(workloadIds);
+                    if (previouslyInstalledWorkloads.Any())
                     {
-                        InstallWorkloads(
-                            workloadIds,
-                            _skipManifestUpdate,
-                            _includePreviews,
-                            offlineCache);
+                        Reporter.WriteLine(string.Format(LocalizableStrings.WorkloadAlreadyInstalled, string.Join(" ", previouslyInstalledWorkloads)).Yellow());
                     }
-                    else
+                    workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                    workloadIds = WriteSDKInstallRecordsForVSWorkloads(workloadIds);
+
+                    if (!_skipManifestUpdate)
```

Comment:
```suggestion
                    if (shouldUpdateWorkloads)
```

There are a couple other places where you use _skipManifestUpdate, too. We need to make sure changes to the value made partway through persist.

Reply:
I've now removed `shouldUpdateWorkloads` entirely.